File: frontend/src/types/paymentReport.ts
export interface PaymentTransaction {
  id:                  string
  period_start:        string
  period_end:          string
  payment_date:        string
  account_id:          number | null
  customer_name:       string
  payment_method:      string | null
  card_last_4:         number | null
  amount:              string
  convenience_fee:     string
  status:              string | null
  reason_code:         string | null
  payment_origin:      string | null
  collector:           string | null
  reference_number:    string | null
  notes:               string | null
  refund_amount:       string | null
  refund_date:         string | null
  refund_initiated_by: string | null
  imported_at:         string
  created_at:          string
  updated_at:          string
}

export interface PaymentKpis {
  total_payments:     number
  total_amount:       string
  total_fees:         string
  total_collected:    string
  total_refunds:      string
  avg_payment_amount: string
}

export interface CollectionStat {
  id:                 string
  period_start:       string
  period_end:         string
  collector:          string
  payments_count:     number
  payments_amount:    string
  autopay_created:    number
  promise_sent:       number
  promise_confirmed:  number
  messages_sent:      number
  notes_count:        number
  waived_fees_count:  number
  waived_fees_amount: string
  worked:             number
  imported_at:        string
}

export interface PaymentByCollector {
  collector:    string
  count:        number
  total_amount: string
}

export interface PaymentByMethod {
  payment_method: string
  count:          number
  total_amount:   string
}

export interface PaymentPage {
  items:  PaymentTransaction[]
  total:  number
  page:   number
  limit:  number
  pages:  number
}

export interface PaymentDateRange {
  from: string  // YYYY-MM-DD
  to:   string  // YYYY-MM-DD
}

File: backend/app/schemas/payment_report.py
from __future__ import annotations

import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class PaymentTransactionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id:                  uuid.UUID
    period_start:        date
    period_end:          date
    payment_date:        datetime
    account_id:          int | None
    customer_name:       str
    payment_method:      str | None
    card_last_4:         int | None
    amount:              Decimal
    convenience_fee:     Decimal
    status:              str | None
    reason_code:         str | None
    payment_origin:      str | None
    collector:           str | None
    reference_number:    str | None
    notes:               str | None
    refund_amount:       Decimal | None
    refund_date:         datetime | None
    refund_initiated_by: str | None
    imported_at:         datetime
    created_at:          datetime
    updated_at:          datetime


class PaymentKpisOut(BaseModel):
    total_payments:     int
    total_amount:       Decimal
    total_fees:         Decimal
    total_collected:    Decimal
    total_refunds:      Decimal
    avg_payment_amount: Decimal


class CollectionStatOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id:                uuid.UUID
    period_start:      date
    period_end:        date
    collector:         str
    payments_count:    int
    payments_amount:   Decimal
    autopay_created:   int
    promise_sent:      int
    promise_confirmed: int
    messages_sent:     int
    notes_count:       int
    waived_fees_count: int
    waived_fees_amount:Decimal
    worked:            int
    imported_at:       datetime


class PaymentByCollectorOut(BaseModel):
    collector:    str
    count:        int
    total_amount: Decimal


class PaymentByMethodOut(BaseModel):
    payment_method: str
    count:          int
    total_amount:   Decimal
